vm-wasmi/src/code_gen.rs

```diff
@@ -1,21 +1,57 @@
 use core::mem;
 
-use alloc::string::String;
-use alloc::{vec, vec::Vec};
+use alloc::{
+    string::{String, ToString},
+    vec,
+    vec::Vec,
+};
 use cosmwasm_std::{ContractResult, Empty, Response};
 use serde::Serialize;
 use wasm_instrument::parity_wasm::{
     builder,
     elements::{FuncBody, Instruction, Instructions, Local, ValueType},
 };
 
+pub enum PredefinedFunctions {
+    Allocate,
+    Instantiate,
+    Execute,
+    Migrate,
+    Deallocate,
+    Query,
+    Reply,
+    DummyFn,
+    InterfaceVersion8,
+    /// This indicates the beginning of the user defined functions
+    UserDefined,
+}
+
+impl ToString for PredefinedFunctions {
+    fn to_string(&self) -> String {
+        match self {
+            PredefinedFunctions::Allocate => "allocate",
+            PredefinedFunctions::Instantiate => "instantiate",
+            PredefinedFunctions::Execute => "execute",
+            PredefinedFunctions::Migrate => "migrate",
+            PredefinedFunctions::Deallocate => "deallocate",
+            PredefinedFunctions::Query => "query",
+            PredefinedFunctions::Reply => "reply",
+            PredefinedFunctions::DummyFn => "dummy_fn",
+            PredefinedFunctions::InterfaceVersion8 => "interface_version_8",
+            PredefinedFunctions::UserDefined => "_",
+        }
+        .into()
+    }
+}
+
```

Comment:
Why not using the existing `AsFunctionName`?

Reply:
Do you mean `AsFunctionName` as a replacement to `PredefinedFunctions`, or having `trait AsFunctionName`?